could not compile polars-arrow v0.42.0: the trait bound [buffer::immutable::Buffer]: Default is not satisfied

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

1. download https://github.com/MarcoGorelli/polars-plugins-tutorial
2. cargo build

### Log output

```shell
🍹 Building a mixed python/rust project
🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.8
🐍 Using CPython 3.9 at D:\Anaconda\envs\py39\python.exe to generate to link bindings (With abi3, an interpreter is only required on windows)
Compiling cfg-if v1.0.0
Compiling once_cell v1.19.0
Compiling libm v0.2.8
Compiling crossbeam-utils v0.8.20
Compiling either v1.13.0
Compiling libc v0.2.158
Compiling allocator-api2 v0.2.18
Compiling psm v0.1.21
Compiling getrandom v0.2.15
Compiling windows_x86_64_msvc v0.36.1
Compiling windows_x86_64_msvc v0.52.6
Compiling stacker v0.1.16
Compiling thiserror v1.0.63
Compiling windows-targets v0.52.6
Compiling windows-sys v0.36.1
Compiling equivalent v1.0.1
Compiling simdutf8 v0.1.4
Compiling bitflags v2.6.0
Compiling static_assertions v1.1.0
Compiling crossbeam-epoch v0.9.18
Compiling raw-cpuid v11.1.0
Compiling smartstring v1.0.1
Compiling bytemuck v1.17.0
Compiling target-features v0.1.6
Compiling crossbeam-deque v0.8.5
Compiling smallvec v1.13.2
Compiling memmap2 v0.7.1
Compiling scopeguard v1.2.0
Compiling bytes v1.7.1
Compiling lock_api v0.4.12
Compiling rayon-core v1.12.1
Compiling parking_lot_core v0.9.10
Compiling multiversion-macros v0.7.4
Compiling byteorder v1.5.0
Compiling atoi_simd v0.15.6
Compiling parking_lot v0.12.3
Compiling rayon v1.10.0
Compiling dyn-clone v1.0.17
Compiling streaming-iterator v0.1.9
Compiling ryu v1.0.18
Compiling ethnum v1.5.0
Compiling itoa v1.0.11
Compiling fast-float v0.2.0
Compiling strength_reduce v0.2.4
Compiling zerocopy v0.7.35
Compiling rand_core v0.6.4
Compiling num-traits v0.2.19
Compiling serde v1.0.209
Compiling array-init-cursor v0.2.0
Compiling ahash v0.8.11
Compiling planus v0.3.1
Compiling multiversion v0.7.4
Compiling xxhash-rust v0.8.12
Compiling pyo3-build-config v0.22.2
Compiling chrono v0.4.38
Compiling polars-error v0.42.0
Compiling memchr v2.7.4
Compiling ppv-lite86 v0.2.20
Compiling hashbrown v0.14.5
Compiling aho-corasick v1.1.3
Compiling regex-syntax v0.8.4
Compiling rand_chacha v0.3.1
Compiling indexmap v2.4.0
Compiling rand v0.8.5
Compiling polars-utils v0.42.0
Compiling polars-arrow-format v0.1.0
Compiling windows-sys v0.52.0
Compiling regex-automata v0.4.7
Compiling polars-arrow v0.42.0
Compiling rand_distr v0.4.3
Compiling home v0.5.9
Compiling glob v0.3.1
Compiling percent-encoding v2.3.1
Compiling pyo3-ffi v0.22.2
Compiling pyo3-macros-backend v0.22.2
Compiling recursive v0.1.1
Compiling regex v1.10.6
Compiling argminmax v0.6.2
error[E0277]: the trait bound [buffer::immutable::Buffer]: Default is not satisfied
--> C:\Users\roar.cargo\registry\src\rsproxy.cn-8f6827c7555bfaf8\polars-arrow-0.42.0\src\compute\cast\binary_to.rs:205:58
|
205 | return BinaryViewArray::try_new(datatype, views, Arc::default(), from.validity().cloned())
| ^^^ the trait Default is not implemented for [buffer::immutable::Buffer], which is required by std::sync::Arc<_, _>: Default
|
= help: the following other types implement trait Default:
&[T]
&mut [T]
[T; 0]
[T; 10]
[T; 11]
[T; 12]
[T; 13]
[T; 14]
and 27 others
= note: required for std::sync::Arc<[buffer::immutable::Buffer]> to implement Default

Compiling fallible-streaming-iterator v0.1.9
Compiling pyo3 v0.22.2
Compiling streaming-decompression v0.1.2
Compiling num-integer v0.1.46
Compiling base64 v0.22.1
Compiling parquet-format-safe v0.2.4
Compiling num-bigint v0.4.6
For more information about this error, try rustc --explain E0277.
error: could not compile polars-arrow (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile polars-arrow (lib) due to 1 previous error
💥 maturin failed
Caused by: Failed to build a native library through cargo
Caused by: Cargo build finished with "exit code: 101": "cargo" "rustc" "--message-format" "json-render-diagnostics" "--manifest-path" "D:\Project\playground\quant\polars_plugin\ops\Cargo.toml" "--lib"
```


### Issue description

I can compile and run my own rust native program that uses polars v0.42.0 as dependency sucessfully.
I tried to use older version of polars to work around, modify the cargo.toml and then removed the cargo.lock then cargo build, but cargo still compiles with polars-arrow v0.42.0:
polars = { version = "0.41.0", default-features = false [}]([url](url))
polars-arrow = { version = "0.41.0", default-features = false }
My question is: Why cargo do not follow my setting? is there any way I can force cargo to use specified version of polars-arrow to compile?

the cargo.toml is attached
[cargo-toml.txt](https://github.com/user-attachments/files/16738606/cargo-toml.txt)



### Expected behavior

compile without errors

### Installed versions

my env: windows10, vs2022, py39, polars 0.42.0,  rust 1.79.0.

--- a/README.md
+++ b/README.md
@@ -217,7 +217,7 @@ improvements point to the `main` branch of this repo.
 polars = { git = "https://github.com/pola-rs/polars", rev = "<optional git tag>" }
 ```
 
-Requires Rust version `>=1.79`.
+Requires Rust version `>=1.80`.
 
 ## Contributing
 

